br/pkg/storage/storage.go

```diff
@@ -198,3 +205,29 @@ func New(ctx context.Context, backend *backuppb.StorageBackend, opts *ExternalSt
 		return nil, errors.Annotatef(berrors.ErrStorageInvalidConfig, "storage %T is not supported yet", backend)
 	}
 }
+
+// copy from `http.defaultTransportDialContext`
+func defaultTransportDialContext(dialer *net.Dialer) func(context.Context, string, string) (net.Conn, error) {
+	return dialer.DialContext
+}
+
+// Different from `http.DefaultTransport`, set the `MaxIdleConns` and `MaxIdleConnsPerHost`
+// to the actual request concurrency to reuse tcp connection as much as possible.
+func GetDefaultHttpClient(concurrency uint) *http.Client {
+	return &http.Client{
+		// copy from `http.DefaultTransport`
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: defaultTransportDialContext(&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}),
+			ForceAttemptHTTP2:     true,
+			MaxIdleConns:          int(concurrency),
+			MaxIdleConnsPerHost:   int(concurrency),
+			IdleConnTimeout:       90 * time.Second,
```

Comment:
Can we avoid to hard code them but copy attributes from http.DefaultTransport

Reply:
the `http.DefaultTransport` is a interface type( interface `RoundTripper` ), so we only can force convert back to Transport and ignore the possible error.
And it has some fields, such as sync.Mutex and map, I'm worried that the copy leaves behind the risk of golang updating.

Reply:
I also like @BornChanger 's idea. We can add an unit test to detect the future implementation change of interface `RoundTripper` , and use `Transport.Clone` to correctly handle its private fields.